tiledb/sm/metadata/metadata.cc

```diff
@@ -107,8 +108,9 @@ Status Metadata::generate_uri(const URI& array_uri) {
   RETURN_NOT_OK(uuid::generate_uuid(&uuid, false));
 
   std::stringstream ss;
-  ss << "__" << timestamp_range_.first << "_" << timestamp_range_.second << "_"
-     << uuid;
+  ss << "__" << std::setfill('0') << std::setw(13) << timestamp_range_.first
```

Comment:
What is the purpose and justification of this change?

Reply:
To fix filename time sequence sorting order issue when setting fixed timestamps that occupy less than max # of digits (without leading zeroes).

This bug that does not show up in prior general usage because timestamps from current time period and forward will always be at max number of possible digits (no leading zeroes), and the existing tests that set fixed timestamps (apparently) never exceeded single digits.

(fragment/metadata) Filenames are sorted according to -textual- order, and when setting fixed timestamps that may have varying numbers of digits (0 ... 20 for example), filenames textually sorted will not be in the correct time-sequence order unless they are all forced to have the same number of digits (i.e. _10_ will sort before _2_, or [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,12,13,14,15,16,17,18,19,20] will sort as [ 0,1,10,11,12,13,14,15,16,17,18,19,2,20,3,4,5,6,7,8,9] or something like that), hence force all timestamps in filenames to have max # of digits.